### PR TITLE
[OTX] Add sample code and test codes for action classification

### PR DESCRIPTION
**Pull request by jaegukhyun**
This PR includes adding sample code and test codes for action classification task.
The newly added test codes only cover training and inference task for action classification.
After enable nncf and openvino task of action classification, test codes for them will be added

### CODE REVIEW

File: data/custom_action_recognition/custom_dataset/train_list_rawframes.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+0 49 0
```
Comment (sungmanc) [inline data/custom_action_recognition/custom_dataset/train_list_rawframes.txt:2]: How about adding the meaning of nubmers on top of this file?

File: otx/algorithms/action/tasks/inference.py [hunk 2]
```diff
@@ -174,7 +184,7 @@ def dummy_dump_saliency_hook(model, inp, out):
 
         metric = None
         metric_name = self._recipe_cfg.evaluation.final_metric
-        if inference_parameters.is_evaluation:
+        if inference_parameters and inference_parameters.is_evaluation:
```
Comment (sungmanc) [inline otx/algorithms/action/tasks/inference.py]: I assume you want to check whether the variable named `inference_parameters` existed.
Then, should we check the existence of the variable first?

```
if inference_parameters: 
    if inference_parameters.is_evaluation:  
        metric = ...  
```

File: tests/integration/cli/action/test_action_classification.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
+"""Tests for MPA Class-Incremental Learning for object detection with OTX CLI"""
```
Comment (sungmanc) [inline tests/integration/cli/action/test_action_classification.py]: Need to update

File: tests/integration/api/action/test_api_action_classification.py [hunk 4]
```diff
@@ -0,0 +1,239 @@
...
+from otx.api.entities.task_environment import TaskEnvironment
+from otx.api.entities.train_parameters import TrainParameters
+from otx.api.usecases.tasks.interfaces.export_interface import ExportType
+from otx.api.utils.shape_factory import ShapeFactory
+from otx.cli.datasets import get_dataset_class
+from otx.cli.utils.io import generate_label_schema
+from tests.test_suite.e2e_test_system import e2e_pytest_api
+from tests.unit.api.test_helpers import generate_random_annotated_image
+
+DEFAULT_ACTION_TEMPLATE_DIR = osp.join("otx/algorithms/action/configs", "classification", "x3d")
```
Comment (sungmanc) [inline tests/integration/api/action/test_api_action_classification.py:43]: Currently, test for only `x3d` ?

File: tests/integration/cli/action/test_action_classification.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+TT_STABILITY_TESTS = os.environ.get("TT_STABILITY_TESTS", False)
+if TT_STABILITY_TESTS:
+    default_template = parse_model_template(
+        os.path.join("otx/algorithms/action/configs", "classification", "x3d", "template.yaml")
+    )
+    templates = [default_template] * 100
+    templates_ids = [template.model_template_id + f"-{i+1}" for i, template in enumerate(templates)]
+else:
+    templates = Registry("otx/algorithms/action").filter(task_type="ACTION_CLASSIFICATION").templates
```
Comment (sungmanc) [inline tests/integration/cli/action/test_action_classification.py:38]: proper indentation is needed

**Review by harimkang — COMMENTED** [review]
LGTM, but this PR need pre-commit test result ("tox -re pre-commit") and api & cli test result in local.

File: tests/integration/api/action/test_api_action_classification.py (hunk 4 above)
Comment (jaegukhyun) [reply tests/integration/api/action/test_api_action_classification.py:43]: Yes, currently X3D is the only backbone for action classification

File: tests/integration/cli/action/test_action_classification.py (hunk 5 above)
Comment (jaegukhyun) [reply tests/integration/cli/action/test_action_classification.py:38]: I cannot see any indentation error. Could you write more specifically?

Comment (sungmanc) [reply tests/integration/cli/action/test_action_classification.py:38]: Ah, sorry, I mis-looked

**Review by harimkang — APPROVED** [review]
LGTM!

**Review by sungmanc — APPROVED** [review]
LGTM

**Pull request merged**